### PR TITLE
[SYCLomatic] Fix memcpy data error of bindless image.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp [hunk 1]
```diff
@@ -289,13 +292,36 @@ static inline sycl::event
...
-  const auto src_extend = sycl::range<3>(src.get_pitch(), src.get_y(), 1);
+  const auto src_extend =
+      sycl::range<3>(src.get_pitch() / ele_size, src.get_y(), 1);
   const auto dest_offset = sycl::range<3>(dest_id[0], dest_id[1], dest_id[2]);
   return q.ext_oneapi_copy(src.get_data_ptr(), src_offset, src_extend,
                            dest->get_handle(), dest_offset, dest->get_desc(),
                            copy_extend);
 }
+
+static inline sycl::event
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpct/bindless_images.hpp:755]: Why we need `static` here?

File: clang/runtime/dpct-rt/include/dpct/memory.hpp [hunk 2]
```diff
@@ -90,6 +90,10 @@ static sycl::event
 dpct_memcpy(const pitched_data src, const sycl::id<3> &src_id,
             image_mem_wrapper *dest, const sycl::id<3> &dest_id,
             const sycl::range<3> &copy_extend, sycl::queue q);
+static sycl::event
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpct/memory.hpp:93]:
```suggestion
inline sycl::event
```

File: clang/runtime/dpct-rt/include/dpct/memory.hpp [hunk 3]
```diff
@@ -668,17 +672,9 @@ dpct_memcpy(sycl::queue &q, const memcpy_parameter &param) {
...
-    host_buffer buf(param.size.size(), q, event_list);
-    to.set_data_ptr(buf.get_ptr());
-    experimental::detail::dpct_memcpy(param.from.image_bindless, param.from.pos,
-                                      to, sycl::id<3>(0, 0, 0), param.size, q);
-    from.set_data_ptr(buf.get_ptr());
-    event_list.push_back(experimental::detail::dpct_memcpy(
-        from, sycl::id<3>(0, 0, 0), param.to.image_bindless, param.to.pos,
-        param.size, q));
-    return event_list;
+    return {experimental::detail::dpct_memcpy(
```
Comment (yihanwg) [inline clang/runtime/dpct-rt/include/dpct/memory.hpp:675]: Does the braces can be removed?

**Review by yihwang-nv — APPROVED** [review]
LGTM

**Review by ziranzha — APPROVED** [review]

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp (hunk 1 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/bindless_images.hpp:755]: Keep the same as other `dpct_memcpy`.

**Review by ziranzha — APPROVED** [review]

**Pull request merged**